common/src/main/java/com/wynntils/commands/LootrunCommand.java

```diff
@@ -0,0 +1,430 @@
+                                        .withStyle(ChatFormatting.GREEN),
+                                false);
+                file.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        return 1;
+    }
+
+    private int recordLootrun(CommandContext<CommandSourceStack> context) {
+        LootrunUtils.register();
+        if (LootrunUtils.recording == null) {
+            LootrunUtils.recording = new LootrunUncompiled(new ArrayList<>(), new HashSet<>(), new ArrayList<>(), null);
+            LootrunUtils.recordingInformation = new RecordingInformation();
+            context.getSource()
+                    .sendSuccess(
+                            new TranslatableComponent(
+                                    "feature.wynntils.lootrunUtils.recordStart",
+                                    new TextComponent("/lootrun record")
+                                            .withStyle(ChatFormatting.UNDERLINE)
+                                            .withStyle((style) -> style.withClickEvent(
+                                                    new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/lootrun record")))),
+                            false);
+        } else {
+            LootrunUtils.lootrun = LootrunUtils.recordingCompiled;
+            LootrunUtils.uncompiled = LootrunUtils.recording;
+            LootrunUtils.recording = null;
+            LootrunUtils.recordingCompiled = null;
+            LootrunUtils.recordingInformation = null;
+            context.getSource()
+                    .sendSuccess(
+                            new TranslatableComponent(
+                                            "feature.wynntils.lootrunUtils.recordStop1",
+                                            new TextComponent("/lootrun clear")
+                                                    .withStyle(ChatFormatting.UNDERLINE)
+                                                    .withStyle((style) -> style.withClickEvent(new ClickEvent(
+                                                            ClickEvent.Action.RUN_COMMAND, "/lootrun clear"))))
+                                    .withStyle(ChatFormatting.RED)
+                                    .append("\n")
+                                    .append(new TranslatableComponent(
+                                                    "feature.wynntils.lootrunUtils.recordStop2",
+                                                    new TextComponent("/lootrun save <name>")
+                                                            .withStyle(ChatFormatting.UNDERLINE)
+                                                            .withStyle((style) -> style.withClickEvent(new ClickEvent(
+                                                                    ClickEvent.Action.SUGGEST_COMMAND,
+                                                                    "/lootrun save "))))
```

Comment:
This extra space should not be there.

Reply:
Why? It is meant to be a command suggestion and the command has name as an argument. So it looks like `/lootrun save <name>`

Reply:
Then I misread the formatting string. Even so, would it not be better to have the space in the surrounding localized string?

Reply:
I think you might have missed that the ClickEvent is a SUGGEST_COMMMAND which opens chat and basically puts "/lootrun save ". The space is there so the player only has to type the name

Reply:
You are absolutely right. I missed that. Now it makes sense. Thanks, and sorry :)